plat/xilinx/zynqmp/aarch64/zynqmp_common.c

```diff
@@ -40,10 +40,19 @@
  * configure_mmu_elx() will give the available subset of that,
  */
 const mmap_region_t plat_arm_mmap[] = {
-	{ DEVICE0_BASE, DEVICE0_BASE, DEVICE0_SIZE, MT_DEVICE | MT_RW | MT_SECURE },
-	{ DEVICE1_BASE, DEVICE1_BASE, DEVICE1_SIZE, MT_DEVICE | MT_RW | MT_SECURE },
-	{ CRF_APB_BASE, CRF_APB_BASE, CRF_APB_SIZE, MT_DEVICE | MT_RW | MT_SECURE },
-	{0}
+	{
+		DEVICE0_BASE, DEVICE0_BASE, DEVICE0_SIZE,
+		MT_DEVICE | MT_RW | MT_SECURE | MT_EXECUTE_NEVER,
+	},
+	{
+		DEVICE1_BASE, DEVICE1_BASE, DEVICE1_SIZE,
+		MT_DEVICE | MT_RW | MT_SECURE | MT_EXECUTE_NEVER,
+	},
+	{
+		CRF_APB_BASE, CRF_APB_BASE, CRF_APB_SIZE,
+		MT_DEVICE | MT_RW | MT_SECURE | MT_EXECUTE_NEVER,
+	},
+	{ 0 }
```

Comment:
I wasn't really aware of that. I just looked through the attributes that are implied in MT_DEVICE and didn't see anything regarding the XN. I'll drop this patch then.